### PR TITLE
CONTRIBUTING.md added

### PR DESCRIPTION
**Pull request by priyanshnama**
I have added CONTRIBUTING.md file to help people in sending pull request and contribute to this.
#3 
Do Suggest changes if any.

### CODE REVIEW

**Comment by priyanshnama** [comment]
@garg3133 tried adding this line to `template.md`

### Wants to add new repo in the list checkout [CONTRIBUTING.md](CONTRIBUTING.md) for the same

but i doubt this would be picked up by python file as i have added it in end.
I have not included this change in the pr as of now.

**Comment by garg3133** [comment]
@priyanshnama How about splitting the template into two files? `template-top.md` and `template-bottom.md`.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Contributing
+
+## Getting Started
+
+1. Submit an issue describing your proposed change.
```
Comment (garg3133) [inline CONTRIBUTING.md]: I think if they are adding a new repo to this list, opening an issue won't make sense.

They should only open issues if they are proposing something new. So, we can make a new section for this something like **Want to propose something?**

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+2. Fork the repository on GitHub.
+3. Clone the fork
+
+## Making Changes
+
+* Create a branch : `git checkout -b branch-name`. Please avoidmworking directly on the `master` branch.
```
Comment (garg3133) [inline CONTRIBUTING.md]: Also, fix the typo here.

File: CONTRIBUTING.md (hunk 1 above)
Comment (priyanshnama) [reply CONTRIBUTING.md]: actually i added that in general sense, if any person encounters any flaws or wants a feature.

**Comment by priyanshnama** [comment]
> @priyanshnama How about splitting the template into two files? `template-top.md` and `template-bottom.md`.

@garg3133 then why to create a new file `template-bottom.md`
we can hardcode the markdown text in python file as a string and use it - as it is just one line of code.
because we will have to bring changes in python file in both ways.

File: CONTRIBUTING.md (hunk 2 above)
Comment (priyanshnama) [reply CONTRIBUTING.md]: sorry for this typo, i shall fix this

**Comment by garg3133** [comment]
@priyanshnama And how about deprecating the use of `template.md` also and bringing all the content from `json` file itself?

**Comment by garg3133** [comment]
Also, I'm thinking of putting repositories in Categories. Any idea how should we make those categories? Like difficulty-wise or language-wise..

**Comment by priyanshnama** [comment]
> @priyanshnama And how about deprecating the use of `template.md` also and bringing all the content from `json` file itself?

### @garg3133 i don't think we should do that.
I fear **it would increase cognitive complexity** of python code.
Moreover, a sperate file **makes us understand how the structure of readme** would be - using only python code won't.
The functionality of python code is really understandable as of now.
Moreover, **making any change in structure of readme would be very easy** till we use `template.md`

**Comment by priyanshnama** [comment]
> Also, I'm thinking of putting repositories in Categories. Any idea how should we make those categories? Like difficulty-wise or language-wise..

@garg3133  yup. we can do this.
I think we should do this language wise.

File: CONTRIBUTING.md (hunk 1 above)
Comment (garg3133) [reply CONTRIBUTING.md]: But it sounds more like they need to open an issue everytime they want to add something, even while adding a new repository.

**Feel free to open an issue if you have something new to propose** or something similar will sound better.

Comment (priyanshnama) [reply CONTRIBUTING.md]: yes... this should be done.. i am pushing a commit for the same

**Pull request merged**